File: samples/conference/payments-bc/src/main/java/com/microsoft/conference/payments/domain/Events/PaymentCompleted.cs
using System;
using ENode.Eventing;

namespace Payments
{
    public class PaymentCompleted : DomainEvent<Guid>
    {
        public Guid OrderId { get; private set; }
        public Guid ConferenceId { get; private set; }

        public PaymentCompleted() { }
        public PaymentCompleted(Payment payment, Guid orderId, Guid conferenceId)
        {
            OrderId = orderId;
            ConferenceId = conferenceId;
        }
    }
}

File: samples/conference/payments-bc/src/main/java/com/microsoft/conference/payments/domain/events/PaymentCompleted.java
package com.microsoft.conference.payments.domain.events;

import com.microsoft.conference.payments.domain.models.Payment;
import lombok.Getter;
import lombok.Setter;
import org.enodeframework.eventing.DomainEvent;

@Getter
@Setter
public class PaymentCompleted extends DomainEvent<String> {
    private String orderId;
    private String conferenceId;

    public PaymentCompleted() {
    }

    public PaymentCompleted(Payment payment, String orderId, String conferenceId) {
        this.orderId = orderId;
        this.conferenceId = conferenceId;
    }
}
